src/OpenTelemetry.Instrumentation.EventCounters/EventCountersInstrumentationOptions.cs

```diff
@@ -14,15 +14,32 @@
 // limitations under the License.
 // </copyright>
 
+using System.Collections.Generic;
+
 namespace OpenTelemetry.Instrumentation.EventCounters;
 
 /// <summary>
-/// EventCounterMetrics Options.
+/// EventCounterListener Options.
 /// </summary>
-public class EventCounterMetricsOptions
+public class EventCountersInstrumentationOptions
 {
     /// <summary>
     /// Gets or sets the subscription interval in seconds.
     /// </summary>
     public int RefreshIntervalSecs { get; set; } = 60;
+
+    /// <summary>
+    /// Gets or sets the names of <c>EventSource</c>s to listen to.
+    /// </summary>
+    public HashSet<string> Sources { get; set; } = new();
+
+    /// <summary>
+    /// Gets or sets the name of <c>EventCounters</c> to listen to.
+    /// </summary>
+    public HashSet<string> Names { get; set; } = new() { "EventCounters" };
+
+    /// <summary>
+    /// Gets the arguments object used for the EventListener.EnableEvents function.
+    /// </summary>
+    public Dictionary<string, string> EnableEventsArguments => new() { { "EventCounterIntervalSec", this.RefreshIntervalSecs.ToString() } };
```

Comment:
And make that private field available to the `EventCountersMetrics.cs` class with a public property?

Reply:
Sorry instead of 'private' what I really intended to convey was 'not part of the public API that 3rd party developers will use'. For example the `internal` keyword would satisfy my goal while still letting EventCounterMetrics invoke it.

 The factoring I would choose personally is moving this property to the EventCounterMetrics type, for example:
```C#
class EventCounterMetrics
{
   ...
   public EventCountersMetrics(EventCountersInstrumentationOptions options)
    {
        this.options = options;

        while (this.preInitEventSources.TryTake(out EventSource source))
        {
            if (this.options.ShouldListenToSource(source.Name))
            {
                this.EnableEvents(source, EventLevel.LogAlways, EventKeywords.None, GetEnableEventsArguments(options));
            }
        }
    }

    private static Dictionary<string,string> GetEnableEventsArguments(EventCountersInstrumentationOptions options) =>
        new() { { "EventCounterIntervalSec", options.RefreshIntervalSecs.ToString() } };
}
```

Not having it be part of the public API is my main recommendation though, regardless whether you use this factoring or the internal keyword or something else.